tools/lint/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"bufio"
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+)
+
+//go:embed outline.txt
+var outline string
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "must provide path to document to check")
+		os.Exit(1)
+	}
+
+	file, err := os.Open(os.Args[1])
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "opening document:", err)
+		os.Exit(1)
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+
+	var lines []string
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
```

Comment:
```suggestion
	fmt.Fprintf(os.Stderr, "expecting %d headers to exist\n", len(lines))
	if err := scanner.Err(); err != nil {
```

Reply:
I'm not a big fan of having functions other than `main()` print directly to stdout (or stderr). What purpose does this print solve?